presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -40,7 +40,7 @@
      */
```

Comment:
Please add tests for queries that were failing previously.

If you are solving https://github.com/prestodb/presto/issues/11276, then please add below query to `io.prestosql.tests.AbstractTestAggregations` or `AbstractTestQueries`:
```
select count(1) from (select sum(field1) from table limit 10) a;
```